File: data/imported/knowledge/cells/cpp/merged/snippets/snippet_045.cpp
TEST(WorkbookMetadataGoldenTests, unreferenced_document_properties_parts_are_ignored)
{
    TempDir temp("golden-workbook-metadata-orphaned-docprops");
    const auto path = temp.Path("workbook-metadata-orphaned-docprops.xlsx");
    auto workbook = CreateWorkbookMetadataWorkbook();
    workbook.Save(path.string());

    RemoveDocumentPropertiesRelationships(path);

    Workbook loaded(path.string());
    EXPECT_EQ("WorkbookCode", loaded.GetProperties().GetCodeName());
    AssertDocumentPropertiesAreDefault(loaded);
}

File: data/imported/knowledge/cells/java/merged/snippets/snippet_045.java
@Test
    void AF_61_filterColumnIndexRoundtripsViaXlsx() throws Exception {
        // Wrap lower-level failures in the library-specific exception flow.
        try (TemporaryDirectory tempDir = new TemporaryDirectory("AutoFilterTest")) {
            String path = tempDir.getPath("af61.xlsx");
            try (Workbook wb = new Workbook()) {
                Worksheet ws = wb.getWorksheets().get(0);
                ws.getAutoFilter().setRange("A1:D1");
                ws.getAutoFilter().getFilterColumns().add(2);
                wb.save(path);
            }
            try (Workbook loaded = new Workbook(path)) {
                assertEquals(1, loaded.getWorksheets().get(0).getAutoFilter().getFilterColumns().getCount());
                assertEquals(2, loaded.getWorksheets().get(0).getAutoFilter().getFilterColumns().get(0).getFieldIndex());
            }
        }
    }